arm-visualstudio/2014-04-01-preview/swagger/Csm.json

```diff
@@ -0,0 +1,818 @@
+        }
+      }
+    },
+    "AccountResourceListResult": {
+      "description": "The response to an account resource list GET request.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Array of resource details.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AccountResource"
+          }
+        }
+      }
+    },
+    "AccountResourceRequest": {
+      "description": "The body of a PUT request to modify a Visual Studio account resource.",
+      "type": "object",
+      "properties": {
+        "accountName": {
+          "description": "The account name.",
+          "type": "string"
+        },
+        "location": {
+          "description": "The Azure instance location.",
+          "type": "string"
+        },
+        "operationType": {
+          "description": "The type of the operation.",
+          "enum": [
+            "unknown",
+            "create",
+            "update",
+            "link"
+          ],
+          "x-ms-enum": {
+            "name": "AccountResourceRequestOperationType"
+          }
+        },
+        "properties": {
+          "description": "The custom properties of the resource.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "x-ms-client-flatten": true
+        },
+        "tags": {
+          "description": "The custom tags of the resource.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "CheckNameAvailabilityParameter": {
+      "description": "The body of a POST request to check name availability.",
+      "type": "object",
+      "properties": {
+        "resourceName": {
+          "description": "The name of the resource to check availability for.",
+          "type": "string"
+        },
+        "resourceType": {
+          "description": "The type of resource to check availability for.",
+          "type": "string"
+        }
+      }
+    },
+    "CheckNameAvailabilityResult": {
+      "description": "The response to a name availability request.",
+      "type": "object",
+      "properties": {
+        "message": {
+          "description": "The message describing the detailed reason.",
+          "type": "string"
+        },
+        "nameAvailable": {
+          "description": "The value which indicates whether the provided name is available.",
+          "type": "boolean"
+        }
+      }
+    },
+    "ExtensionResource": {
+      "description": "The response to an extension resource GET request.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/AccountResource"
+        }
+      ],
+      "properties": {
+        "plan": {
+          "description": "The extension plan that was purchased.",
+          "$ref": "#/definitions/ExtensionResourcePlan"
+        }
+      }
+    },
```

Comment:
Inheriting from Resource instead now that we have a base class makes sense. I'll change that.

I saw the warnings about your other points in the swagger validator but making these changes in the current API version could introduce some service-to-service compat issues for us. Can we leave this as-is for the current 2014 API version and address this (and the other inconsistencies/design issues you've noticed in the RP) for a future release?